### ISSUE TITLE
Configure network programmatically

### ISSUE THREAD

**Issue by bramroelandts** [siemens/meta-iot2050#162]
We are in the process of writing a web interface on top of the IOT2050 with various configuration options, under which one can configure the network settings for both Ethernet ports. We are having difficulties, however, with configuring the network settings manually.

By default, the box uses `NetworkManager`. The `iot2050setup` script uses a GUI provided by `NetworkManager` itself to configure these network settings, if I correctly interpreted the Python source code. We would like to change network settings from `NodeJS`, so using this GUI is not an option. It's possible to interact with `nmcli` through d-bus, but there's only a few obscure open-source libraries, nothing mature enough for production.

To simplify matters, we tried disabling `NetworkManager` and just rely on default Linux networking, by creating and editing files in `/etc/network/interfaces.d/` and using `ifup` and `ifdown`. This approach, however, seems very unstable. We expected it to work smoothly, but all kinds of issues have arisen:
- Sometimes when the DHCP lease expires, the interface does not get a new IP address and the box becomes unreachable. Only way of saving it is logging in via UART or over SSH via the other (statically configured) interface. This happens even though dhclient is running.
- When `eth1` is configured as DHCP and the gateway is connected to an outbound internet connection this way, some programs (like npm) do not get through to the internet, presumably because routing is incorrect. This does not happen however when `eth0` is connected to the internet.
- There's weird things going on with `ifdown` and `ifup`. While `ifconfig` sees both interfaces, running `ifdown eth0` returns errors saying no such interface exists.

In short, the box becomes generally unstable. On a Debian system, it is not expected you have to worry about DHCP leases and routing tables. What would be the best way to configure networking manually, while not breaking the whole system? We'd like to stay as close as possible to what Siemens developed, but can't really figure out what's the best way without experiencing major problems.

Thanks in advance!

**Comment by jan-kiszka** [comment]
Please reopen if you have any follow-up questions.
